zap/src/main/java/org/parosproxy/paros/core/scanner/AppParameter.java

```diff
@@ -0,0 +1,41 @@
+package org.parosproxy.paros.core.scanner;
+
+/**
+ * {@code AppParameter} class wraps the parameters which are used to modify the {@code HttpMessage}.
+ * It is specifically used for updating multiple parameters of {@code HttpMessage}
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class AppParameter {
+    private NameValuePair nameValuePair;
+    private String param;
+    private String value;
+    private AppParamValueType appParamValueType;
+
+    AppParameter(
+            NameValuePair nameValuePair,
+            String param,
+            String value,
+            AppParamValueType appParamValueType) {
+        this.nameValuePair = nameValuePair;
+        this.param = param;
+        this.value = value;
+        this.appParamValueType = appParamValueType;
+    }
+
+    public NameValuePair getNameValuePair() {
+        return nameValuePair;
+    }
+
+    public String getParam() {
```

Comment:
`getName()`

Reply:
I think that these method names should align with earlier Api's parameter names. Else it will confuse users. WDYT?

Reply:
The main consumers of this API would be the variants, which know about the name (`NameValuePair#getName()`).

Reply:
hmm right as a consumer will be using the builder. Make sense.